sdk-base/api/sdk-base.api

```diff
@@ -607,17 +607,20 @@ public final class com/mapbox/maps/plugin/LocationPuck3D : com/mapbox/maps/plugi
 	public fun <init> (Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;)V
 	public fun <init> (Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;Ljava/util/List;)V
 	public fun <init> (Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;Ljava/util/List;Ljava/util/List;)V
-	public synthetic fun <init> (Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;Ljava/util/List;Ljava/util/List;ILkotlin/jvm/internal/DefaultConstructorMarker;)V
+	public fun <init> (Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;Ljava/util/List;Ljava/util/List;Z)V
+	public synthetic fun <init> (Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;Ljava/util/List;Ljava/util/List;ZILkotlin/jvm/internal/DefaultConstructorMarker;)V
 	public final fun component1 ()Ljava/lang/String;
 	public final fun component2 ()Ljava/util/List;
 	public final fun component3 ()F
 	public final fun component4 ()Ljava/util/List;
 	public final fun component5 ()Ljava/lang/String;
 	public final fun component6 ()Ljava/util/List;
 	public final fun component7 ()Ljava/util/List;
-	public final fun copy (Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;Ljava/util/List;Ljava/util/List;)Lcom/mapbox/maps/plugin/LocationPuck3D;
-	public static synthetic fun copy$default (Lcom/mapbox/maps/plugin/LocationPuck3D;Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;Ljava/util/List;Ljava/util/List;ILjava/lang/Object;)Lcom/mapbox/maps/plugin/LocationPuck3D;
+	public final fun component8 ()Z
+	public final fun copy (Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;Ljava/util/List;Ljava/util/List;Z)Lcom/mapbox/maps/plugin/LocationPuck3D;
+	public static synthetic fun copy$default (Lcom/mapbox/maps/plugin/LocationPuck3D;Ljava/lang/String;Ljava/util/List;FLjava/util/List;Ljava/lang/String;Ljava/util/List;Ljava/util/List;ZILjava/lang/Object;)Lcom/mapbox/maps/plugin/LocationPuck3D;
```

Comment:
Is binary compatibility something that we committed to deliver? If I'm not mistaken we don't guarantee backwards compatibility between releases on iOS.

Reply:
As far as I know we try to commit to both binary and source compatibility, but there're some known issues for both cases, which need to be revisited in v11. cc @tobrun

Reply:
> Is binary compatibility something that we committed to deliver? If I'm not mistaken we don't guarantee backwards compatibility between releases on iOS.

Yes, we do our best to follow semver and keep binary compatibility. On Android breaking it could crash the users apps, e.g. Nav SDK depends on some version of Maps, but in the app user can pick newer Maps, gradle will resolve such situation and crash if Maps has some breaking changes.

@pengdev where there different option (without data class modification)? This type of breaking changes seems _acceptable_  to me, but I'd avoid if possible, ofc.

Reply:
The other option is to fetch the 3D puck model layer and use the model layer APIs to change this layer property. However for now, this property seems most relevant to 3D location puck, so added here as experimental feature for the ease of use.

Reply:
Suppose we could use an extension methods instead of adding property to the data class wdyt?